### PR TITLE
Add unit tests for ControllerPublish and ControllerUnpublish requests

### LINKED ISSUE #199: Implement Unit Test for ControllerPublishVolume and ControllerUnpublishVolume
**Is your feature request related to a problem?/Why is this needed**
- [x] ControllerPublishVolume
- [x] ControllerUnpublishVolume

### PR DESCRIPTION
**Pull request by sreis**
**Is this a bug fix or adding new feature?**

Increases test coverage. 

**What is this PR about? / Why do we need it?**

Increases test coverage.

Closes #199 

**What testing is done?** 

Add unit tests for ControllerPublish and ControllerUnpublish requests.

### CODE REVIEW

**Comment by leakingtapan** [comment]
travis is having issue these days:
```
{"message":"Couldn't find a repository matching this job.","error":true}
The command "$GOPATH/bin/goveralls -coverprofile=profile.cov -service=travis-ci" exited with 1.
```

File: pkg/driver/controller_test.go [hunk 1]
```diff
@@ -429,3 +429,204 @@ func TestPickAvailabilityZone(t *testing.T) {
 		})
 	}
 }
+
+func TestControllerPublishVolume(t *testing.T) {
+	fakeCloud := NewFakeCloudProvider()
```
Comment (leakingtapan) [inline pkg/driver/controller_test.go]: Could you create a `mock_cloud` using gomock? With that, we could truly unit test controller service without depending on the fake cloud implementation. 

Here is something similar I did for another driver: https://github.com/aws/aws-fsx-csi-driver/blob/master/pkg/driver/controller_test.go#L224

Comment (sreis) [reply pkg/driver/controller_test.go]: That's a good idea but maybe we can do it in a follow up PR and introduce it in all the required tests. wdyt?

Comment (leakingtapan) [reply pkg/driver/controller_test.go]: Fair enough.

**Comment by leakingtapan** [comment]
@sreis could you rebase. thx

**Comment by sreis** [comment]
@leakingtapan I've rebased PTAL.

File: pkg/driver/controller_test.go (hunk 1 above)
Comment (leakingtapan) [reply pkg/driver/controller_test.go]: #220

**Pull request merged**